src/app/vitessce-grid-utils.js

```diff
@@ -103,20 +104,15 @@ export function createLoaders(datasets, configDescription) {
     };
     dataset.files.forEach((file) => {
       // Fall back to JsonLoader if a loader is not found for the file type.
-      const matchingLoaderClass = fileTypeToLoader[file.fileType] || JsonLoader;
-      let loader;
-      if (Array.isArray(matchingLoaderClass)) {
-        const [DataSourceClass, LoaderClass] = matchingLoaderClass;
-        // Create _one_ BaseLoaderClass instance per URL. Derived loaders share this object.
-        if (!(file.url in dataSources)) {
-          const { url, requestInit } = file;
-          dataSources[file.url] = new DataSourceClass({ url, requestInit });
-        }
-        loader = new LoaderClass(dataSources[file.url], file);
-      } else {
-        // eslint-disable-next-line new-cap
-        loader = new matchingLoaderClass(file);
+      const matchingLoaderClass = fileTypeToLoaderAndSource[file.fileType]
+        || [JsonSource, JsonLoader];
```

Comment:
Perhaps move this logic into `"../loaders/types.js"` and export a function that handles the default behavior.

```javascript
import { getLoaderAndSourceFromFileType } from '../loaders/types';
/* ... */
const matchingLoaderClass = getLoaderAndSourceFromFileType(file.fileType);
```

Reply:
Also, since loaders are now _all_ two step and we removed the `if` block:

```javascript
const [DataSourceClass, LoaderClass] = getLoaderAndSourceFromFileType(file.fileType);
```